Sources/Hyperspace/Service/Transport/TransportService.swift

```diff
@@ -42,19 +43,25 @@ extension TransportService: Transporting {
             self?.networkActivityController?.stop()
             self?.handle(data: data, response: response, error: error, for: HTTP.Request(urlRequest: request), completion: completion)
         }
-        
-        tasks[request] = task
-        task.resume()
-        
-        networkActivityController?.start()
+
+        queue.async { [weak self] in
+            self?.tasks[request] = task
+            task.resume()
+
+            self?.networkActivityController?.start()
```

Comment:
In the interest of symmetry, we should probably either serially `start` and `stop`, or neither. If I remember correctly, `NetworkActivityController` already has an internal queue, so I'd recommend we leave it out of this `queue` altogether.

Reply:
Fair point, I'll move it out of the queue.